examples/connectivity/aws-connection/main.tf

```diff
@@ -31,14 +31,18 @@ resource "equinix_ecx_l2_connection" "example" {
   authorization_key = var.aws_account_id
 }
 
-resource "equinix_ecx_l2_connection_accepter" "example" {
-  connection_id = equinix_ecx_l2_connection.example.id
-  access_key    = var.aws_access_key
-  secret_key    = var.aws_secret_key
+resource "aws_dx_connection" "example" {
+  name      = "tf-aws-dot1q"
+  bandwidth = "50Mbps"
+  location  = var.aws_location_code
```

Comment:
It would be nice if there was a way to match the location code with an Equinix location for the purposes of these examples. I could not find one so I took the path of least resistance and suggested that the user define this value in a variable.

Reply:
I'm also creating the connection in this example.  It seems we have the option to instead use an existing connection via https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/dx_connection. For the purposes of the example, I think either approach is fine. (assuming this code works - I was not able to verify this. I'm hoping @ocobleseqx  may be able to.)

Reply:
Since the connection in AWS is configured through the 'equinix_exc_l2_connection' resource the connection id should be referenceable in that resource. This was already discussed here #22 .

AWS dx_connection datasource could be an alternative but there may be an issue, AWS allows creating multiple direct connect connections with the same name but the aws_dx_connection datasource doesn't allow more filters than [name](https://github.com/hashicorp/terraform-provider-aws/blob/main/internal/service/directconnect/connection_data_source.go#L52)

Reply:
There is a convention to use a singular name or a plural name in data sources depending on whether the resource name returns a map or array of results or a single result. When a resource is expected to return a single result, the provider will typically throw an error if more than one results match the selection criteria.

I don't know if this AWS follows that convention, but I think if this approach works at all, it is sufficient for the example. If it helps to avoid an incorrect example, we could simplify further with `var.aws_connection_id` instead of creating or looking up the AWS DX Connection.

Reply:
From #22
> When creating an AWS Virtual Interface within the Direct Connect service, a "Connection ID" attribute must be used. In the absence of AWS Data Source for 'aws_dx_connection' resource, there's no way to fetch this attribute from AWS.

We now have https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/dx_connection available.

> Equinix Terraform provider allows creating an "equinix_ecx_l2_connection" resource and relies on ECX API to create the Direct Connect connection. ECX API L2 connection object includes actionDetails list that includes the AWS connection_id:

It seems we may need to revisit https://github.com/equinix/terraform-provider-equinix/issues/26.  

@pfeiffermj proposed `connection_id     = 
  equinix_ecx_l2_connection_details.details.actionDetails.actionRequiredData.ELEMENT.awsConnectionId.value` which @abudilov also suggested in https://github.com/equinix/terraform-provider-equinix/issues/22.

Issues 22 and 26 discuss an `actionDetails` element of the ECX API which contains a list of maps with field values to populate AWS credentials and retrieve AWS connection details.  While improved access to this attribute (Terraform providers should match the upstream API), this API pattern has been revoked by the successor API, ECX APIv3:

>  V3 removes the need for a service provider to approve a deleted connection and immediately deletes the connection and stops billing when this action is performed by a user

More details are provided on https://developer.equinix.com/docs/ecx-v3-apis-migration and https://developer.equinix.com/docs/ecx-connect-aws.

> As of January 5, 2020, Equinix stopped supporting Equinix Fabric V2 APIs, and after July 5, 2020, all previous versions of Equinix Fabric APIs have been deprecated.

It seems we may require a more comprehensive overhaul.

For now, I believe it would be sufficient to find a way to discover the AWS Connection ID.